packages/shared/src/components/modals/SteppedModal.tsx

```diff
@@ -3,37 +3,77 @@ import React, {
   KeyboardEvent,
   MouseEvent,
   ReactElement,
+  useContext,
   useState,
 } from 'react';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+import useAuthForms from '../../hooks/useAuthForms';
+import { AuthEventNames } from '../../lib/auth';
+import { logout } from '../../lib/user';
+import AuthOptions, { AuthDisplay } from '../auth/AuthOptions';
+import { DiscardAuthModal } from '../auth/common';
 import { Button } from '../buttons/Button';
 import TabContainer, { Tab } from '../tabs/TabContainer';
 import { ModalProps, StyledModal } from './StyledModal';
 
 interface SteppedModalProps extends ModalProps {
+  trigger: string;
+  skippable?: boolean;
   isLastStepLogin?: boolean;
+  isFirstStepIntroduction?: boolean;
   onStepChange?: (stepBefore: number, stepNow: number) => void | Promise<void>;
   onFinish?: () => void | Promise<void>;
+  onAuthSuccess?: (isLogin?: boolean) => void | Promise<void>;
 }
 
 function SteppedModal({
```

Comment:
This is actually a really cool generic component, but we now tie it down to the auth, is it not better to push auth as the last step from outside?

Just thinking this would make a super easy adjustment and make this way more flexible?
But if it's a lot of work or hassle don't worry about it.

Reply:
Yep, good point. My only worry was the generic modal would have no idea to overrule the footer of next and previous. I am actually torn if those things should be handled outside as well and just pass a state for the max and current step. WDYT?

Though the auth is optional, you will pass a certain parameter to when it should be the last step or not at all.

Reply:
If it's too much work just leave it, we'll deal with it once we need it :)